extensions/git/src/main.ts

```diff
@@ -20,6 +20,7 @@ import { GitProtocolHandler } from './protocolHandler';
 import { createGitExtension } from './api/extension';
 import * as path from 'path';
 import * as fs from 'fs';
+import * as which from 'which';
```

Comment:
We probably don't need `which` for this to work. We especially can't use it if the user hasn't placed Code in the PATH. You're going to have to find another way to do this.

Reply:
I need some help for that. 
I wanted to use `require('electron').app.getPath('exe')` ([doc](https://github.com/electron/electron/blob/v2.0.11/docs/api/app.md#appgetpathname) ) but it fails with `Cannot find module 'electron'` and indeed, I don't have it in the `node_modules` folder.

Does I miss something? Should I add 'electron' as a dev dependency?

Should I find another way? Use the ExtensionContext to pass the value?

Reply:
One can't require `electron` in the extension host. We also can't use `require('electron').app.getPath('exe')` because for Windows (for example) the right launcher to use is within the `bin` folder. There might not be an easy way to get to it, as an extension, actually.

Reply:
😕 So my initial approach to look at the vscode exe in the PATH is the better option?

Or you see something else to try?

Reply:
I actually don't think we have a good solution for doing this as an extension. We probably need some extra API to expose the location of the CLI.